### PR TITLE
 ENH: Easy compilation of LISF at tier-2 and tier-1

### PR DESCRIPTION
**Pull request by mbechtold**
This PR changes the default configuring routine and adds the modules list and slurm scripts to compile LDT and LIS on the two HPC systems Tier-1 (currently hortense) and Tier-2 (genius) of the Vlaams Supercomputing Center. The system is automatically recognized as well. So master-kul and every branch from it will be easy to compile without manually changing files. 
The idea is to keep the compilation settings up to date when new clusters become available. 

To Do: Shorten outdated Github documentation on ldt and lis compilation. One can basically refer to the README_COMPILATION files in lis and ldt.

### CODE REVIEW

**Review by lbusschaert — APPROVED** [review]
This is great Michel, thank you for putting time into this! I tested the compilation on genius following the README file and it all works. I left a minor comment suggesting to use different processes to speed up the compilation.

File: ldt/compile_ldt_genius.slurm [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#SBATCH --cluster=genius
+#SBATCH -o compile_ldt_genius_log.txt
+#SBATCH -e compile_ldt_genius_out.txt
+
+####
+# for larger lis simulations that need more than 6 cores you should add #SBATCH --exclude=r23i13n23
+
+cd $SLURM_SUBMIT_DIR
+source KUL_LDT_modules
+./compile
```
[the remark this answers is not in the document]
Comment (mbechtold) [reply ldt/compile_ldt_genius.slurm]: Great suggestion. Added.

File: lis/README_COMPILATION.TXT [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# 2024/03/04: Michel Bechtold
+# To compile lis, run the following steps
+cd LISF/lis
+# load the modules (right ones for each system will be automatically selected)
+source KUL_LIS_modules
+# run the configure and accept all defaults or change to 'debug' compilation if wanted
+./configure 
+# submit either the hortense or genius job scripts
+# hortense (tier 1)
+qsub compile_lis_hortense.qsub
```
Comment (zdenkoheyvaert) [inline lis/README_COMPILATION.TXT:10]: file extension should be `.pbs`.

File: ldt/README_COMPILATION.TXT [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# 2024/03/04: Michel Bechtold
+# To compile ldt, run the following steps
+cd LISF/ldt
+# load the modules (right ones for each system will be automatically selected)
+source KUL_LDT_modules
+# run the configure and accept all defaults or change to 'debug' compilation if wanted
+./configure 
+# submit either the hortense or genius job scripts
+# hortense (tier 1)
+qsub compile_ldt_hortense.qsub
```
Comment (zdenkoheyvaert) [inline ldt/README_COMPILATION.TXT:10]: file extension should be `.pbs`.

File: lis/README_COMPILATION.TXT (hunk 2 above)
Comment (mbechtold) [reply lis/README_COMPILATION.TXT]: Fixed.

File: ldt/README_COMPILATION.TXT (hunk 3 above)
Comment (mbechtold) [reply ldt/README_COMPILATION.TXT]: Fixed.

File: ldt/arch/Config.pl [hunk 4]
```diff
@@ -407,15 +417,26 @@
...
+      }
       $lib = "/lib/";
       $inc_hdf4=$sys_hdf4_path.$inc;
       $lib_hdf4=$sys_hdf4_path.$lib;
    }
    else {
       print "--------------ERROR---------------------\n";
       print "Please specify the HDF4 path using\n";
-      print "the LDT_HDF4 variable.\n";
+      print "the LIS_HDF4 variable.\n";
```
Comment (gdelannoy) [inline ldt/arch/Config.pl:439]: not sure why LDT_ is explicitly changed to LIS_ in a file for ldt

**Review by gdelannoy — APPROVED** [review]
Looks great. Perhaps fix one error message, if needed (LDT vs LIS).

**Review by zdenkoheyvaert — APPROVED** [review]
Tested on Tier-1 and works perfectly, much easier than before! Thanks.

It had been a while since I submitted a PBS script (usually work interactively on OnDemand now), which made me realize I also needed to run `module swap cluster/dodrio/cpu_milan` for this to work. But I think we can assume people have this in their `.bashrc`, so no need to add it here I think.

File: ldt/arch/Config.pl (hunk 4 above)
Comment (mbechtold) [reply ldt/arch/Config.pl]: Fixed.

**Pull request merged**